### ISSUE TITLE
Convert optional properties to options 

### ISSUE THREAD

**Issue by Borewit** [Borewit/strtok3#124]
If `maybeless` flag is switched on, the `offset`, `length` and `position` should be provided as well.

 `offset`, `length` are pretty straightforward, `position` is bit tricky.

Applies to:
*   [tokenizer.readBuffer()](https://github.com/Borewit/strtok3/blob/master/README.md#method-tokenizerreadbuffer)
*   [tokenizer.peekBuffer()](https://github.com/Borewit/strtok3/blob/master/README.md#method-tokenizerpeekbuffer)

for example:
```js
await tokenizer.peekBuffer(buffer, 0, bytesRead, tokenizer.position, true);
```

suggest to change to:
```js
await tokenizer.peekBuffer(buffer, 0, bytesRead, {maybeless: true});
```
Not sure to move `offset` (target buffer) and `bytesRead` to the options as well.

**Comment by sindresorhus** [comment]
I think the `maybeless` name could be improved. It's not clear what it means. Maybe something like `strictLength` which could be set to `false`?

> Not sure to move offset (target buffer) and bytesRead to the options as well.

I think `offset` should also move into the options object. Regarding `length`, it's not clear from the docs what the behavior is when it's not specified?

**Comment by Borewit** [comment]
Appreciate your feedback @sindresorhus, very useful, I will try to make it more clear.

**Comment by Borewit** [comment]
Implemented in [v6.0.0](https://github.com/Borewit/strtok3/releases/tag/v6.0.0)
